vmm/src/vm.rs

```diff
@@ -854,20 +854,20 @@ impl Vm {
 
         cfg_if::cfg_if! {
             if #[cfg(feature = "tdx")] {
+                let mut tdx_opt = 0; // KVM_X86_LEGACY_VM
+                if tdx_enabled {
+                    tdx_opt = 1; // KVM_X86_TDX_VM
+                }
                 let vm = hypervisor
-                    .create_vm_with_type(if tdx_enabled {
-                        1 // KVM_X86_TDX_VM
-                    } else {
-                        0 // KVM_X86_LEGACY_VM
-                    })
+                    .create_vm_with_type(tdx_opt)
```

Comment:
It is better to use `u64::from(tdx_enabled)` and add a comments for the two VM types.

Reply:
Fixed.